src/umpire/strategy/DynamicPoolHeuristic.hpp

```diff
@@ -31,7 +31,19 @@ class DynamicPool;
    *
    * \return True if all memory in pool is releaseable.
    */
-  bool heuristic_all_allocations_are_releaseable( const strategy::DynamicPool& dynamic_pool );
+  bool heuristic_100_percent_releaseable( const strategy::DynamicPool& dynamic_pool );
+
+  /*!
+   * \brief Return true if 90% of pool is releaseable
+   *
+   * When 90% has been deallocated back to the pool, this heuristic
+   * will return true.
+   *
+   * \param dynamic_pool The dynamic pool object in question.
+   *
+   * \return True if 90% memory in pool is releaseable.
+   */
+  bool heuristic_90_percent_releaseable( const strategy::DynamicPool& dynamic_pool );
```

Comment:
@davidbeckingsale, can you think of a way that I could make this function more generic?  I'd like to parameterize the percentage to use rather than creating a separate callback for each percentage.

Reply:
You can have it be a function that generates a lambda. e.g. it takes the percentage, and returns a function that takes a DynamicPool. Does that make sense?

Reply:
Great suggestion!  Done.